File: anmeldetool-server/src/main/java/org/ztv/anmeldetool/models/TiTuEnum.java
package org.ztv.anmeldetool.models;

public enum TiTuEnum {
	Ti("Turnerin"), Tu("Turner"), Alle("Gemeinsamer Wettkampf");

	private String beschreibung;

	TiTuEnum(String beschreibung) {
		this.beschreibung = beschreibung;
	}

	public String getBeschreibung() {
		return beschreibung;
	}

	public boolean equals(String name) {
		return this.toString().equals(name);
	}

	public boolean isTurner() {
		return TiTuEnum.Alle.equals(this) || TiTuEnum.Tu.equals(this);
	}
}

File: anmeldetool-web/src/main/web/src/app/core/model/TiTuEnum.ts
export enum TiTuEnum {
  Ti = 'Turnerin',
  Tu = 'Turner',
  Alle = 'Gemeinsamer Anlass',
}

export const TiTuValues = Object.values(TiTuEnum);

export const parseTiTuEnum = (value?: string): TiTuEnum | undefined => {
  if (!value) {
    return undefined;
  }

  if (value in TiTuEnum) {
    return TiTuEnum[value as keyof typeof TiTuEnum];
  }

  return TiTuValues.find((tiTu) => tiTu === value);
};

export const isTiTuEnumEqual = (a?: string, b?: string): boolean => parseTiTuEnum(a) === parseTiTuEnum(b);
